libtcmu_log.c

```diff
@@ -239,12 +241,168 @@ void tcmu_dbg_scsi_cmd_message(const char *funcname, int linenr, const char *fmt
 	va_end(args);
 }
 
+static int append_output(log_output_fn output_fn, log_close_fn close_fn, void *data,
+                         int pri, int dest, const char *name)
+{
+	char *ndup = NULL;
+	struct log_output output;
+
+	if (output_fn == NULL)
+	    return -1;
+
+	if (dest == TCMU_LOG_TO_FILE) {
+		if (name == NULL)
+			return -1;
+		ndup = strdup(name);
+		if (ndup == NULL)
+			return -1;
+	}
+
+	output.output_fn = output_fn;
+	output.close_fn = close_fn;
+	output.data = data;
+	output.priority = pri;
+	output.dest = dest;
+	output.name = ndup;
+
+	darray_append(logbuf->outputs, output);
+
+	return 0;
+}
+
+static int output_to_syslog(int pri, const char *timestamp,
+                            const char *str, void *data)
+{
+	/* convert tcmu-runner private level to system level */
+	if (pri > TCMU_LOG_DEBUG)
+		pri = TCMU_LOG_DEBUG;
+	syslog(pri, "%s", str);
+	return strlen(str);
+}
+
+static void close_syslog(void *data)
+{
+	closelog();
+}
+
+static void close_fd(void *data)
+{
+	int fd = (intptr_t) data;
+	close(fd);
+}
+
+static int create_syslog_output(int pri, const char *ident)
+{
+	open_syslog(ident, 0 ,0);
+	if (append_output(output_to_syslog, close_syslog, NULL,
+			  pri, TCMU_LOG_TO_SYSLOG, ident) < 0) {
+		closelog();
+		return -1;
+	}
+	return 0;
+}
+
+static const char *loglevel_string(int priority)
+{
+	switch (priority) {
+		case TCMU_LOG_ERROR:
```

Comment:
I know you just copied the style from the existing code, but since you are moving it could you also just fix the indention to match the other code. 

switch (priority) {
case TCMU_LOG_ERROR:

So no tab for the case line.